# Lab 7

## HW6 Due TODAY

I briefly reviewed the submissions, and it seems that some people missed the bins. Please double check your solutions, so that you have the correct bin argument inside the histograms.

## Today's agenda

- Conditional Probability
- Simulation
- Birthday problem
- Law of large numbers
- Central limit theorem

## Brief overview of conditional probability

- People who aren't comfortable with conditional probability/Bayes rule should consult Lecture 10 slides

What is conditional probability?

- We want to find out the probability of the event A given that the event B occurred

$P(A|B)$ is probability of A given that the event B occurred

<img src="Image/Bayes.png" width=400>

$$
P(A|B) = \frac{P(A\cap B)}{P(B)}=\frac{Y}{Y+Z}
$$

### P(A|B) vs P(B|A)?

Beginners in conditional probability might have some difficulty remembering which one is which. Read it aloud. $P(A|B)$ is the probability of $A$ given $B$, thus you have the first one ($A$) on your left and the second one ($B$) on your right and put a bar $|$ between them.

# Bayes' Rule:

There are some theorems that you must memorize in statistics. Bayes rule is one of them.

$$
P(A|B) = \frac{P(B|A)P(A)}{P(B)}
$$

The codes below comes directly from HW4 solution

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('Data/NHANES.csv')

In [3]:
df = df[['Race3','Age','BPSysAve','BPDiaAve']]
df = df.dropna()

In [4]:
def function(row):
    if row['BPDiaAve'] < 80 and row['BPSysAve'] < 120:
        return 'Healthy'
    elif row['BPDiaAve'] < 80 and row['BPSysAve'] < 130:
        return 'Elevated'
    elif row['BPDiaAve'] < 90 and row['BPSysAve'] < 140:
        return 'Stage1'
    elif row['BPDiaAve'] <= 120 and row['BPSysAve'] <= 180:
        return 'Stage2'
    elif row['BPDiaAve'] >= 180 or row['BPSysAve'] >= 120:
        return 'Hypertensive Crisis'
    else:
        return 'NaN'

In [5]:
df['Blood Pressure'] = df.apply(function, axis=1)
table = pd.crosstab(index=df["Blood Pressure"], columns=df["Race3"]) 
table

Race3                Asian  Black  Hispanic  Mexican  Other  White
Blood Pressure                                                    
Elevated                26     81        32       50     29    433
Healthy                153    238       158      256     68   1453
Hypertensive Crisis      0      6         0        0      0     19
Stage1                  36     97        56       50     11    506
Stage2                  27     63        27       25     14    367

In [6]:
table['Asian_per'] = table['Asian'] / sum(table['Asian'])
table['Black_per'] = table['Black'] / sum(table['Black'])
table['Hispanic_per'] = table['Hispanic'] / sum(table['Hispanic'])
table['Mexican_per'] = table['Mexican'] / sum(table['Mexican'])
table['Other_per'] = table['Other'] / sum(table['Other'])
table['White_per'] = table['White'] / sum(table['White'])
table = table[['Asian_per','Black_per','Hispanic_per','Mexican_per','Other_per','White_per']]
table

Race3                Asian_per  Black_per  Hispanic_per  Mexican_per  \
Blood Pressure                                                         
Elevated              0.107438   0.167010      0.117216     0.131234   
Healthy               0.632231   0.490722      0.578755     0.671916   
Hypertensive Crisis   0.000000   0.012371      0.000000     0.000000   
Stage1                0.148760   0.200000      0.205128     0.131234   
Stage2                0.111570   0.129897      0.098901     0.065617   

Race3                Other_per  White_per  
Blood Pressure                             
Elevated              0.237705   0.155868  
Healthy               0.557377   0.523038  
Hypertensive Crisis   0.000000   0.006839  
Stage1                0.090164   0.182145  
Stage2                0.114754   0.132109

What does White_per 0.523 represents?

1. $P(Healthy\cup White)$
2. $P(Healthy\cap White)$
3. $P(Healthy|White)$
4. $P(White|Healthy)$

It's 3. $P(Healthy|White)$. This probability represents the probability a person is healthy, given that the person is white. This was obtained by:

$$
P(Healthy|White) = \frac{\text{Healthy White}}{\text{White}}=\frac{P(Healthy\cap White)}{P(White)}
$$